packages/rum-core/test/common/slugify.spec.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import slugifyUrl from '../../src/common/slugify'
+
+describe('Slug URL', () => {
+  const validate = (before, after, depth) => {
+    expect(slugifyUrl(before, depth)).toBe(after)
+  }
+
+  it('accept depth parameter', () => {
+    validate('/a/b/c/d', '/a/b/**')
+    validate('/a/b/c/d', '/a/**', 1)
+  })
+
+  it('handle trailing slash', () => {
+    validate('/a/', '/a')
+    validate('/a', '/a')
+    validate('/', '/')
+  })
+
+  it('handle query param', () => {
+    validate('/a/b/?id=hello', '/a/b?{query}')
+    validate('/?foo=bar&bar=baz', '/?{query}')
```

Comment:
Should we remove the query string altogether? I think it's not useful in the transaction name.

Reply:
Hmm, but we need a differentiation between search landing page and search query page in ecommerce context.

```
/search
vs
/search?{query}
```